streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimestampedStore.java

```diff
@@ -383,7 +383,9 @@ public KeyValue<Bytes, byte[]> makeNext() {
 
         @Override
         public synchronized void close() {
-            openIterators.remove(this);
+            if (closeCallback != null) {
```

Comment:
Don't we require that we always have a registered `closeCallback`? If yes, it seems invalid that it's `null` and having this check might mask a bug, and thus we should rather not have it, but let it crash with a NPE? (Also elsewhere.)

Reply:
Unclear whether we want to require that a closeCallback is always registered in general, but it is true that for these two specific classes (RocksDbIterator and RocksDBDualCFIterator), we do want to require that a closeCallback is always set. I've updated the code to reflect this.